app/scripts/modules/core/src/pipeline/config/triggers/Triggers.tsx

```diff
@@ -54,7 +54,7 @@ export function Triggers(props: ITriggersProps) {
         <NotificationsPageContent {...props} />
       </PageSection>
       <PageSection pageKey="description" label="Metadata" noWrapper={true}>
-        <MetadataPageContent {...props} />
+        <MetadataPageContent {...props} key={revertCountKLUDGE} />
```

Comment:
remount MetadataPageContent when the revertCount changes.

Reply:
TBH I am not sure why we don't just remount the entire Triggers component when we revert, but this is all so fragile, I don't want to rock the boat any further in this PR.

Reply:
Remounting everything on revert does seem to be the only reasonable thing to do.